packages/menu/src/menu.vue

```diff
@@ -205,24 +205,19 @@ export default defineComponent({
       }
 
       if (props.router && router && hasIndex) {
-        routeToItem(item, error => {
-          activeIndex.value = oldActiveIndex
-          if (error) {
-            // vue-router 3.1.0+ push/replace cause NavigationDuplicated error
-            // https://github.com/ElemeFE/element/issues/17044
-            if (error.name === 'NavigationDuplicated') return
-            console.error(error)
-          }
-        })
-      }
-    }
+        let route = item.route || item.index
+        router?.push(route)
+          .then(navigationResult => {
+            if (navigationResult && navigationResult.type !== 16) {
```

Comment:
I'm sorry that we have to avoid using magic numbers so that can you make 16 as a local variable like:

```ts
const DUPLICATE_ROUTE = 16

if (navigationResult && navigationResult.type !== DUPLICATE_ROUTE) {
  // ... other code
}
```
🙏

Reply:
This is the one last change request, after you change this we should be able to merge this PR. Thank you so much for collaborating. 🙇

Reply:
ok了:smile:

Reply:
太棒了 谢谢配合！